flink-runtime/src/main/java/org/apache/flink/runtime/state/SnapshotResult.java

```diff
@@ -0,0 +1,83 @@
+package org.apache.flink.runtime.state;/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import org.apache.flink.util.ExceptionUtils;
+
+/**
+ *
+ */
+public class SnapshotResult<T extends StateObject> implements StateObject {
```

Comment:
I'm wondering whether it wouldn't make sense to have a `SnapshotResult` and a subclass `SnapshotResultWithLocalSnapshot`. The latter has an additional field for the `taskLocalSnapshot`. That way it's much clearer how to initialize these structures. Ideally the arguments are both not null.

Reply:
That way we would also get rid of all the null checks.